crates/sui-types/src/messages_checkpoint.rs

```diff
@@ -286,93 +289,48 @@ impl SignedCheckpoint {
 // or other authenticated data structures to support light
 // clients and more efficient sync protocols.
 
-#[derive(Clone, Debug, Serialize, Deserialize)]
-pub struct CertifiedCheckpoint {
-    pub checkpoint: CheckpointSummary,
-    signatures: Vec<(AuthorityName, AuthoritySignature)>,
-}
+pub type CertifiedCheckpointSummary = CheckpointSummaryEnvelope<AuthorityQuorumSignInfo>;
 
-impl CertifiedCheckpoint {
+impl CertifiedCheckpointSummary {
     /// Aggregate many checkpoint signatures to form a checkpoint certificate.
     pub fn aggregate(
-        signed_checkpoints: Vec<SignedCheckpoint>,
+        signed_checkpoints: Vec<SignedCheckpointSummary>,
         committee: &Committee,
-    ) -> Result<CertifiedCheckpoint, SuiError> {
+    ) -> Result<CertifiedCheckpointSummary, SuiError> {
         fp_ensure!(
             !signed_checkpoints.is_empty(),
             SuiError::from("Need at least one signed checkpoint to aggregate")
         );
 
-        let certified_checkpoint = CertifiedCheckpoint {
-            checkpoint: signed_checkpoints[0].checkpoint.clone(),
-            signatures: signed_checkpoints
-                .into_iter()
-                .map(|v| (v.authority, v.signature))
-                .collect(),
+        let certified_checkpoint = CertifiedCheckpointSummary {
+            summary: signed_checkpoints[0].summary.clone(),
+            auth_signature: AuthorityQuorumSignInfo {
+                // TODO: Add proper epoch info.
+                epoch: 0,
+                signatures: signed_checkpoints
+                    .into_iter()
+                    .map(|v| (v.auth_signature.authority, v.auth_signature.signature))
+                    .collect(),
+            },
         };
 
         certified_checkpoint.verify(committee)?;
         Ok(certified_checkpoint)
     }
 
     pub fn signatory_authorities(&self) -> impl Iterator<Item = &AuthorityName> {
-        self.signatures.iter().map(|(name, _)| name)
+        self.auth_signature.signatures.iter().map(|(name, _)| name)
     }
 
     /// Check that a certificate is valid, and signed by a quorum of authorities
     pub fn verify(&self, committee: &Committee) -> Result<(), SuiError> {
-        // Note: this code is nearly the same as the code that checks
-        // transaction certificates. There is an opportunity to unify this
-        // logic.
-
-        let mut weight = 0;
-        let mut used_authorities = HashSet::new();
-        for (authority, _) in self.signatures.iter() {
-            // Check that each authority only appears once.
-            fp_ensure!(
-                !used_authorities.contains(authority),
-                SuiError::CertificateAuthorityReuse
-            );
-            used_authorities.insert(*authority);
-            // Update weight.
-            let voting_rights = committee.weight(authority);
-            fp_ensure!(voting_rights > 0, SuiError::UnknownSigner);
-            weight += voting_rights;
-        }
-        fp_ensure!(
```

Comment:
Where is this check done, and is it done for >1/3 as it should here, rather than >2/3 as for many other certs we handle?